doc/user/python.md

```diff
@@ -139,8 +139,7 @@ methods.
 The key space can be worked through via python iterators.
 
 ```
->>> for key in dd: print(key, dd[key])
->>> for key in dd: print(key, dd[key])
+>>> for key, value in enumerate(dd): print(key, value)
```

Comment:
Sorry, typed my previous comment too quickly. What I meant was that we would actually need implementation support for enumerate to work like this. Consider a normal python dictionary:
```
>>> d = {'key_a': 'val_a', 'key_b': 'val_b', 'key_c': 'val_c'}
>>> for i, key in enumerate(d):
...     print(i, key)
...
0 key_a
1 key_b
2 key_c
>>> for i, (key, val) in enumerate(d.items()):
...     print(key, val)
...
key_a val_a
key_b val_b
key_c val_c
```
So really, to make this efficient we would want to override dd.items() (or define bitems(), etc.) to return an iterator to bulk pre-fetch several values, but still yield 1 value on each iteration.
Then, we could do this:
```
for i, (key, val) in enumerate(dd.items()):
```
Or probably more simply:
```
for key, value in dd.items() # implementation actually prefetches multiple values
```
By default I don't think dict.items() is an iterator, so it gets all values. So for now we should stick with:
```
for key in dd: print(key, dd[key])
```

Sorry for the confusion :)

Reply:
ok, got it.